src/nvim/ex_getln.c

```diff
@@ -1005,6 +1009,12 @@ static int command_line_execute(VimState *state, int key)
   }
 
   s->do_abbr = true;             // default: check for abbreviation
+
+  // if in number mode, key is alphabet
+  if (s->firstc == '#' && ASCII_ISALPHA(s->c)) {
```

Comment:
isn't it better to check if it is a digit?

Reply:
well, then this check should be moved to after special chars are checked, and just before a char is inserted into the cmdline string.